core/src/test/scala/unit/kafka/log/UnifiedLogTest.scala

```diff
@@ -3951,18 +3953,18 @@ class UnifiedLogTest {
     assertEquals(10, log.logSegments.size())
 
     {
-      val deletable = log.deletableSegments(
-        (segment: LogSegment, _: Option[LogSegment]) => segment.baseOffset <= 5)
-      val expected = log.nonActiveLogSegmentsFrom(0L).asScala.filter(segment => segment.baseOffset <= 5).toList
-      assertEquals(6, expected.length)
-      assertEquals(expected, deletable.toList)
+      val deletable = new util.ArrayList(
```

Comment:
Do we actually need this copy?

Reply:
This is strictly not needed as it returns `ArrayList` in this test's successful case scenario but the same method can return `EmptyList` instance also in other case. I wanted the test to fail with details about the elements instead of type mismatch.